src/HomaRouter.sol

```diff
@@ -24,17 +25,45 @@ contract HomaRouter is BaseRouter {
         _instructions = instructions;
     }
 
+    function toEvmAddress(bytes32 addr) private pure returns (address) {
+        bytes32 prefix = bytes32(uint256(0x65766d3a00000000000000000000000000000000000000000000000000000000));
+        bool checkPrefix = addr & prefix == prefix;
+        if (!checkPrefix) {
+            return address(0);
+        }
+
+        bytes32 suffix = bytes32(uint256(0x000000000000000000000000000000000000000000000000ffffffffffffffff));
+        bool checkSuffix = addr & suffix == 0;
+        if (!checkSuffix) {
+            return address(0);
+        }
+
+        // convert addr[4..24] to address
+        address result = address(bytes20(addr << 32));
+        return result;
+    }
+
+    function transfer(ERC20 token, bytes32 addr) private {
+        address recipient = toEvmAddress(addr);
+        if (recipient == address(0)) {
+            // Substrate account
+            // This will fail if token is not a native token.
+            // That means user is doing something wrong and will lose their tokens.
```

Comment:
should we add something like a `emergencyWithdraw` method to avoid token loss?

Reply:
yeah I can add that